### PR TITLE
feat(aws-s3): Adding Intelligent Tiering to Bucket

### LINKED ISSUE #16191: (s3): Configure intelligent tiering in bucket constructor
Specifies the S3 Intelligent-Tiering configuration for an Amazon S3 bucket. It is supported by `aws_cdk.aws_s3.CfnBucket`but is not supported by  `aws_cdk.aws_s3.Bucket`.

### Use Case
I use `aws_cdk.aws_s3.Bucket` and I want to use this feature in CDK. There is work around  but  if want to have my bucket configuration in a centralize point.

### Proposed Solution
Add the `intelligent_tiering_configurations` to `aws_cdk.aws_s3.Bucket`.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by makrsmark**
Allows users to add Intelligent Tiering to their s3 buckets. Modelling of CfnBucket was done before.

Fixes #16191

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-s3/README.md [hunk 1]
```diff
@@ -480,15 +480,28 @@ by deploying with CDK version `1.126.0` or later **before** switching this value
...
 ```
+
+## Intelligent Tiering
+
+[Intelligent Tiering](https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html) can be configured to enable fast, easy, and secure transfers of files over long distances:
+
+```ts
+    new s3.Bucket(this, 'MyBucket', {
+   intelligentTieringConfigurations: [{
+      name: 'foo',
```
Comment (otaviomacedo) [inline packages/@aws-cdk/aws-s3/README.md:503]: I think you could include other props here to make the usefulness of this feature more evident, like `prefix`, `archiveAccessTierTime`.

**Review by otaviomacedo — COMMENTED** [review]
Looks good! Just a minor comment on the documentation.

File: packages/@aws-cdk/aws-s3/README.md (hunk 1 above)
Comment (makrsmark) [reply packages/@aws-cdk/aws-s3/README.md:503]: updated

**Review by otaviomacedo — APPROVED** [review]

**Comment by makrsmark** [comment]
```ERR! 503 Service Unavailable - GET https://registry.npmjs.org/https-proxy-agent - Service Unavailable```

I think this is an issue with npm registry - other PRs are failing for the same reason but different packages

**Comment by makrsmark** [comment]
@otaviomacedo could you please re-run the CodeBuild approval as npmjs should be operational again?

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**